Beams/Logic/vtkSlicerIECTransformLogic.cxx

```diff
@@ -298,15 +333,145 @@ bool vtkSlicerIECTransformLogic::GetTransformBetween(CoordinateSystemIdentifier
   {
-    vtkMRMLTransformNode::GetTransformBetweenNodes(
-      this->GetTransformNodeBetween(Collimator, Gantry), nullptr, outputTransform );
+    std::vector< CoordinateSystemIdentifier > toFrameVector(toFramePath.size());
+    std::vector< CoordinateSystemIdentifier > fromFrameVector(fromFramePath.size());
+
+    std::copy( toFramePath.begin(), toFramePath.end(), toFrameVector.begin());
+    std::copy( fromFramePath.begin(), fromFramePath.end(), fromFrameVector.begin());
+
+    outputTransform->Identity();
+    outputTransform->PostMultiply();
+    for ( size_t i = 0; i < fromFrameVector.size() - 1; ++i)
+    {
+      CoordinateSystemIdentifier parent, child;
+      child = fromFrameVector[i];
+      parent = fromFrameVector[i + 1];
+      vtkMRMLLinearTransformNode* fromTransform = this->GetTransformNodeBetween( child, parent);
+
+      vtkDebugMacro("GetTransformBetween: Checking affine transformation " 
+        << "\"" << this->CoordinateSystemsMap[child] << "\" -> " 
+        << "\"" << this->CoordinateSystemsMap[parent] << "\"");
+
+      if (fromTransform)
+      {
+        vtkNew<vtkMatrix4x4> mat;
+        fromTransform->GetMatrixTransformToParent(mat);
+        outputTransform->Concatenate(mat);
+
+        vtkDebugMacro("GetTransformBetween: Transform node from frame " 
+          << "\"" << this->CoordinateSystemsMap[parent] << "\" to frame " 
+          << "\"" << this->CoordinateSystemsMap[child] << "\" is valid");
+      }
+      else
+      {
+        vtkErrorMacro("GetTransformBetween: Transform node from frame " 
+          << "\"" << this->CoordinateSystemsMap[parent] << "\" to frame " 
+          << "\"" << this->CoordinateSystemsMap[child] << "\" is invalid");
+        return false;
+      }
+    }
+
+    for ( size_t i = 0; i < toFrameVector.size() - 1; ++i)
+    {
+      CoordinateSystemIdentifier parent, child;
+      parent = toFrameVector[i];
+      child = toFrameVector[i + 1];
+      vtkMRMLLinearTransformNode* toTransform = this->GetTransformNodeBetween( child, parent);
+
+      vtkDebugMacro("GetTransformBetween: Checking affine transformation " 
+        << "\"" << this->CoordinateSystemsMap[parent] << "\" -> " 
+        << "\"" << this->CoordinateSystemsMap[child] << "\"");
+
+      if (toTransform)
+      {
+        vtkNew<vtkMatrix4x4> mat;
+        toTransform->GetMatrixTransformFromParent(mat);
+        mat->Invert();
+        outputTransform->Concatenate(mat);
+
+        vtkDebugMacro("GetTransformBetween: Transform node to frame " 
+          << "\"" << this->CoordinateSystemsMap[child] << "\" from " 
+          << "\"" << this->CoordinateSystemsMap[parent] << "\" is valid");
+      }
+      else
+      {
+        vtkErrorMacro("GetTransformBetween: Transform node from frame " 
+          << "\"" << this->CoordinateSystemsMap[child] << "\" to frame " 
+          << "\"" << this->CoordinateSystemsMap[parent] << "\" is invalid");
+        return false;
+      }
+    }

```

Comment:
I think it would make sense to either log an error here (as an else for the failed path retrievals), or in those functions themselves (GetPathToRoot and GetPathFromRoot).